homeassistant/components/switchgrid/calendar.py

```diff
@@ -0,0 +1,92 @@
+"""Support for Switchgrid Calendar platform."""
+
+import datetime
+
+from homeassistant.components.calendar import CalendarEntity, CalendarEvent
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.device_registry import DeviceEntryType, DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+import homeassistant.util.dt as dt_util
+
+from .const import DOMAIN
+from .coordinator import SwitchgridCoordinator
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the calendar platform for entity."""
+    coordinator: SwitchgridCoordinator = hass.data[DOMAIN][config_entry.entry_id]
+    async_add_entities([SwitchgridCalendarEntity(coordinator, config_entry)])
+
+
+class SwitchgridCalendarEntity(
+    CoordinatorEntity[SwitchgridCoordinator], CalendarEntity
+):
+    """A calendar entity holding Switchgrid Electric LoadShedding events."""
+
+    _attr_has_entity_name = True
+    _attr_unique_id = "switchgrid_events"
```

Comment:
The unique id only needs to be unique among all switchgrid calendars, there's no need to prefix with switchgrid
```suggestion
    _attr_unique_id = "events"
```